examples/mplot3d/lorenz_attractor.py

```diff
@@ -30,33 +30,23 @@ def lorenz(x, y, z, s=10, r=28, b=2.667):
     x_dot = s*(y - x)
     y_dot = r*x - y - x*z
     z_dot = x*y - b*z
-    return x_dot, y_dot, z_dot
+    return np.array([x_dot, y_dot, z_dot])
 
 
 dt = 0.01
 num_steps = 10000
 
-# Need one more for the initial values
-xs = np.empty(num_steps + 1)
-ys = np.empty(num_steps + 1)
-zs = np.empty(num_steps + 1)
-
-# Set initial values
-xs[0], ys[0], zs[0] = (0., 1., 1.05)
-
+xyzs = np.empty((num_steps + 1, 3))  # Need one more for the initial values
+xyzs[0] = (0., 1., 1.05)  # Set initial values
 # Step through "time", calculating the partial derivatives at the current point
 # and using them to estimate the next point
 for i in range(num_steps):
-    x_dot, y_dot, z_dot = lorenz(xs[i], ys[i], zs[i])
-    xs[i + 1] = xs[i] + (x_dot * dt)
-    ys[i + 1] = ys[i] + (y_dot * dt)
-    zs[i + 1] = zs[i] + (z_dot * dt)
-
+    xyzs[i + 1] = xyzs[i] + lorenz(*xyzs[i]) * dt
```

Comment:
I'm not sure how idiomatic we want to be in python - many scientific programmers, moving between languages, are not going to understand what you are doing here with the *xyzs.  Even if it is more typing and even a bit more error prone, I think writing the code out is easier to parse than nice encapsulations like this.

Reply:
Fair point.  Would you think it would be better if lorenz() took a single xyz as sole arg, and did `x, y, z = xyz`?

Reply:
+1 for array in, array out

Reply:
done

Reply:
The `*` is still here?

Reply:
oops, fixed.